File: spring-boot-hibernate/src/main/java/com/example/hibernatebootcamp/service/UserService.java
package com.example.hibernatebootcamp.service;

import com.example.hibernatebootcamp.entity.UserEntity;
import com.example.hibernatebootcamp.repository.UserRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public List<UserEntity> getAllUsers() {
        return userRepository.findAll();
    }

    public UserEntity saveUser(UserEntity userEntity) {
        return userRepository.save(userEntity);
    }
} 

File: spring-boot-kotlin/src/main/kotlin/com/example/revision/service/UserService.kt
package com.example.revision.service

import com.example.revision.entity.UserEntity
import com.example.revision.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(private val userRepository: UserRepository) {

    fun createUser(user: UserEntity): UserEntity = userRepository.save(user)

    fun getUserById(id: String): UserEntity? = userRepository.findById(id).orElse(null)

    fun updateUser(id: String, user: UserEntity): UserEntity? {
        return if (userRepository.existsById(id)) {
            val updatedUser = user.copy(id = id)
            userRepository.save(updatedUser)
        } else null
    }

    fun deleteUser(id: String): Boolean {
        return if (userRepository.existsById(id)) {
            userRepository.deleteById(id)
            true
        } else false
    }

    fun getAllUsers(): List<UserEntity> = userRepository.findAll()

}